extensions/ql-vscode/src/vscode-tests/cli-integration/queries.test.ts

```diff
@@ -100,4 +100,27 @@ describe('Queries', function() {
     }
   });
 
+  // Asserts a fix for bug https://github.com/github/vscode-codeql/issues/733
+  it.only('should restart the database and run a query', async () => {
+    try {
+      await commands.executeCommand('codeQL.restartQueryServer');
+      const queryPath = path.join(__dirname, 'data', 'simple-query.ql');
+      const result = await compileAndRunQueryAgainstDatabase(
+        cli,
+        qs,
+        dbItem,
+        false,
+        Uri.file(queryPath),
+        progress,
+        token
+      );
+
+      // this message would indicate that the databases were not properly reregistered
+      expect(result.result.message).not.to.eq('No result from server');
+      expect(result.options.queryText).to.eq(fs.readFileSync(queryPath, 'utf8'));
+    } catch (e) {
+      console.error('Test Failed');
+      fail(e);
+    }
```

Comment:
Was this for debugging or do you need to keep it?

Reply:
We should keep this. The tests aren't failed if an error is thrown. We need to explicitly throw it here.